### PR TITLE
Replace lift with parameterised error handling

### PR DESCRIPTION
**Pull request by joeldrapper**

### CODE REVIEW

File: test/literal/result.test.rb [hunk 1]
```diff
@@ -1,219 +1,3 @@
 # frozen_string_literal: true
 
-# # frozen_string_literal: true
```
Comment (joeldrapper) [inline test/literal/result.test.rb:3]: Need to replace these tests.

Comment (stevegeek) [reply test/literal/result.test.rb:3]: Want a hand with them?

Comment (joeldrapper) [reply test/literal/result.test.rb:3]: I'm actually stalling because I'm working on GreenDots itself a bit first, trying to improve the failure output. 😅

Comment (stevegeek) [reply test/literal/result.test.rb:3]: Hah ok!

**Pull request closed without merging**